src/Core/src/Platform/Gtk/ImageView.cs

```diff
@@ -9,11 +11,60 @@ namespace Microsoft.Maui.Platform
 
 	public class ImageView : Gtk.Image
 	{
+		// OriginalSizeBuf is added so when we want to resize the image using Pixbuf we can access the
+		// initial Pixbuf and resize that. 
+		Pixbuf? _originalSizeBuf;
+		// Image might be null because it is possible to have a WidthRequest before Pixbuf has a value.
+		// In this case, we'll just save the value, and once Pixbuf is initialized we'll update its size
+		// to have the saved values
+		int? _lastRequestedWidth;
+		int? _lastRequestedHeight;
 
-		public Gdk.Pixbuf? Image
+		public Pixbuf? Image
 		{
 			get => Pixbuf;
-			set => Pixbuf = value;
+			set
+			{
+				_originalSizeBuf = value;
+				if (_lastRequestedHeight is not null || _lastRequestedWidth is not null)
+				{
+					Pixbuf = _originalSizeBuf?.ScaleSimple(
+						_lastRequestedWidth ?? _originalSizeBuf.Width,
+						_lastRequestedHeight ?? _originalSizeBuf.Height,
+						InterpType.Bilinear
+					);
+				}
+				else
+				{
+					Pixbuf = value;
+				}
+			}
+		}
+
+		public void ChangeHeight(int height)
+		{
+			if (Image is null)
+			{
+				_lastRequestedHeight = height;
+			}
+			else
+			{
+				var newBuf = _originalSizeBuf?.ScaleSimple(Pixbuf.Width, height, InterpType.Bilinear);
+				Pixbuf = newBuf;
+			}
+		}
+
+		public void ChangeWidth(int width)
```

Comment:
I think this Change* methods can be internal

Reply:
I tried removing the "public" keyword, but then I got compile errors. So, no they can't.

Reply:
I didnt suggest to remove the public keyword (which means making it private), I explicitly said **internal**

Reply:
sorry I miss-interpreted your message. it's fixed now.